lib/directory-tree.js

```diff
@@ -27,16 +27,18 @@ function directoryTree (path, options, onEachFile) {
 
 	try { stats = FS.statSync(path); }
 	catch (e) { return null; }
-	if (stats.isFile()) {
-		const ext = PATH.extname(path).toLowerCase();
 
-		// Only add files with the provided extensions
-		if (options && 
-			options.extensions && 
-			options.extensions.length && 
-			options.extensions.indexOf(ext) === -1)
-			return null;
+	// Skip if it matches the exclude regex
+	if (options && options.exclude && options.exclude.test(path))
+		return null;  
 
+	if (stats.isFile()) {
+		// Skip if it does not match the include regex
+		if (options && options.includeOnly && !options.includeOnly.test(path))
```

Comment:
@mihnea-dobrescubalaur-hs So I think the best way to do this is to provide an extension `includeOnly` as opposed to folders. The other plugins generally use a regex to match the path pattern to be traversed, which doesn't make sense in this case because we have to start with a root directory. How do you think we should move forward? I think having an extensions regex and a excludes regex should be good enough.